pandas/_libs/index.pyx

```diff
@@ -509,7 +509,7 @@ cdef class PeriodEngine(Int64Engine):
     cdef _get_index_values(self):
         return super(PeriodEngine, self).vgetter()
 
-    cdef void _call_map_locations(self, values):
+    cpdef _call_map_locations(self, values):
```

Comment:
these are still cdef on master, why is this the appropriate fix on 1.0.4?

Reply:
> why is this the appropriate fix on 1.0.4?

#29232 was a clean-up PR which caused a regression. this is reverting those changes. (applied directly to 1.0.x branch and not touching master, see https://github.com/pandas-dev/pandas/pull/34048#issuecomment-625347097)

> these are still cdef on master

that's my question to you in the OP.

> Could there still be issue on master with the exceptions not being caught?

Reply:
> Could there still be issue on master with the exceptions not being caught?

I can't reproduce the bugs from #32409.  I'm guessing you bisected to find the commit that introduced (reintroduced?) the bug, did you find what commit fixed it?

Reply:
yep fixed on master, see https://github.com/pandas-dev/pandas/issues/32409#issuecomment-606627286

but not sure whether the PR that fixed it #31294 should be backported. I can't immediately see the relation. 

I see this PR as the safer option as it's a simple revert.

Reply:
> I see this PR as the safer option as it's a simple revert.

Works for me